src/project.cpp

```diff
@@ -1923,8 +1923,16 @@ Map* Project::addNewMapToGroup(QString mapName, int groupNum, Map *newMap, bool
     if (!existingLayout) {
         mapLayouts.insert(map->layoutId, map->layout);
         mapLayoutsTable.append(map->layoutId);
-        setNewMapBlockdata(map);
-        setNewMapBorder(map);
+        if (importedMap) {
+            map->layout->lastCommitMapBlocks.blocks = new Blockdata;
```

Comment:
Is this history-related code needed?  You should just creating a new map in the same pattern as the existing New Map window, which doesn't need history-related logic.